php/libraries/VisitController.class.inc

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * This file contains the VisitController class.
+ *
+ * PHP Version 7
+ *
+ * @category Controller
+ * @package  Visits
+ * @author   Mélanie Legault <melanie.legault2@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS;
+
+/**
+ * Base class that contains method to access the visit of projects
+ *
+ * @category Controller
+ * @package  Visits
+ * @author   Mélanie Legault <melanie.legault2@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class VisitController
+{
+    /**
+     * The database connection to use
+     *
+     * @var \Database
+     */
+    protected $database;
+
+    /**
+     * The constructor
+     *
+     * @param \Database $database The database connection to use
+     */
+    public function __construct(\Database $database)
+    {
+        if (!$database->isConnected()) {
+            throw new \RuntimeException("VisitController - Database not connected");
+        }
+        $this->database = $database;
+    }
+
+    /**
+     * Retruns a list of configured Visits
+     *
+     * @return array The list of visits
+     */
+    public function getAllVisits(): array
+    {
+        return array_map(
+            function ($row) {
+                return new \LORIS\Visit(
+                    $row['visit']
+                );
+            },
+            $this->database->pselect(
+                'SELECT
+                 v.VisitName as "visit"
+                FROM
+                 visit v
+                ',
+                array()
+            )
+        );
+    }
+
+    /**
```

Comment:
I'm having trouble understanding what this function does.

**with** keyword in the function name suggests that you are adding and ID to the visit object but instead you are querying from the database and retuning a whole new visit with only the ID and not the name... so thats a bit confusing. I'm not sure visits should be in plural, we do assume that the visit name is unique (maybe throw an exception if more than one result exist) and finally the parameter name $visit is not informative, $visitName would be much clearer

it seems like pselectOne is much cleaner than what is currently queried and if you follow my advice above, you can use pselectRow and return a visit oject with an ID and a Name because thats what the visit is, returning just the ID i na visit object having supplied the name is not ideal

Reply:
As the name of the fonction imply, it return a list of all the visit configure in the system as object of type visit.

Reply:
I don't understand what you mean. Where do you see a "with" keyword?

Reply:
`getVisitsWithID` WITHid